x-pack/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_resources_client.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { sha256 } from 'js-sha256';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import type {
+  RuleMigrationResource,
+  RuleMigrationResourceType,
+} from '../../../../../common/siem_migrations/model/rule_migration.gen';
+import type { StoredRuleMigrationResource } from '../types';
+import { RuleMigrationsDataBaseClient } from './rule_migrations_data_base_client';
+
+const BULK_MAX_SIZE = 500 as const;
+
+export class RuleMigrationsDataResourcesClient extends RuleMigrationsDataBaseClient {
+  public async upsert(resources: RuleMigrationResource[]): Promise<void> {
+    const index = await this.getIndexName();
+
+    let resourcesSlice: RuleMigrationResource[];
+    while ((resourcesSlice = resources.splice(0, BULK_MAX_SIZE)).length > 0) {
+      await this.esClient
+        .bulk({
+          refresh: 'wait_for',
```

Comment:
Do you have any concerns with `refresh` on serverless? May not be a big deal since the user will be running this only once or a few times at most...